### ISSUE TITLE
MaskedCausalVisionTransformer ignores masks with gradient checkpointing

### ISSUE THREAD

**Issue by Purin1410** [lightly-ai/lightly#2049]
`MaskedCausalVisionTransformer.forward_features()` ignores its attention mask when gradient checkpointing is enabled. The checkpointed masked output matches an unmasked forward instead of the ordinary masked path.

```python
import torch

from lightly.models.modules.masked_causal_vision_transformer import MaskedCausalVisionTransformer

torch.manual_seed(0)
model = MaskedCausalVisionTransformer(
    img_size=32, patch_size=16, embed_dim=24, depth=2, num_heads=3
).eval()
images = torch.rand(2, 3, 32, 32)
mask = torch.zeros(2, 5, dtype=torch.bool)
mask[:, 1:] = True

expected = model.forward_features(images, mask=mask)
model.set_grad_checkpointing()
actual = model.forward_features(images, mask=mask)

torch.testing.assert_close(actual, expected)
```

Environment: Linux 6.8, Python 3.12.3, PyTorch 2.13.0 CPU.
